state-network.md

```diff
@@ -255,53 +269,297 @@ sedes      := Container(enrs: List[byte_list, max_length=32], payload: byte_list
+
+> TODO: Figure out language for defining which trie nodes from this proof the bridge node must initialize gossip.
+
+> TODO: Determine mechanism for contract code.
+
+
+#### Stage 2A: Neighborhood Trie Node Gossip
+
+
+When individual trie nodes are gossiped they will be transmitted as both the trie node itself, and the additional proof nodes necessary to prove the trie node against a recent state root.
+
+The receiving DHT node will perform *neighborhood* gossip to nearby nodes from their routing table.
+
+
+#### Stage 2B: Recursive Trie Node Gossip
+
+When individual trie nodes are gossiped, the receiving node is responsible for initializing gossip for other trie nodes contained in the accompanying proof.
+
+This diagram illustrates the proof for the trie node under the path `0011`.  
+
+```
+0:                           X
+                            / \
+                          /     \
+                        /         \
+                      /             \
+                    /                 \
+                  /                     \
+                /                         \
+1:             0                           1
+             /   \
+           /       \
+         /           \
+2:      0             1
+       / \
+      /   \
+3:   0     1
+          / \
+4:       0  (1)
+
+```
+
+Note that it also includes the intermediate trie nodes under the paths:
+
+- `000`
+- `001`
+- `00`
+- `01`
+- `0`
+- `1`
+- `X`
+
+The gossip payload for the trie node at `0011` will contain these trie nodes as a proof.  Upon receipt of this trie node and proof, the receiving DHT node will extract the proof for the intermediate node under the path `001` which is the direct parent of the main trie node currently being gossiped.  This proof would be visualized as follows:
+
+
+```
+0:                           X
+                            / \
+                          /     \
+                        /         \
+                      /             \
+                    /                 \
+                  /                     \
+                /                         \
+1:             0                           1
+             /   \
+           /       \
+         /           \
+2:      0             1
+       / \
+      /   \
+3:   0    (1)
+
+4:
+
+```
+
+The DHT node would then *initialize* gossip for the trie node under the path `001`, using this new slimmed down proof to anchor to the same state root.  This process continues until it reaches the state root where it naturally terminates.  We refer to this as *recursive trie node* gossip.
+
+#### Stage 2C: Initialization Of Leaf Proof Gossip
+
+When receiving an individual trie node which represents a "leaf" node in the trie, the combined leaf node and proof are equivalent to a *"leaf proof"*.  The DHT node receiving this data in the form of "trie node" data is responsible for *initializing gossip* for the same data as a *"leaf proof"*.
```

Comment:
it's not clear her which kind of gossip the DHT nodes initiate

Reply:
Maybe clarifying that these aren't stages of a state machine, too? But yes, I think this is propagating the Stage 3 data here.